### ISSUE TITLE
Crasher when running with parallelism enabled

### ISSUE THREAD

**Issue by lihaoyi** [lihaoyi/mill#913]
This is happening non-deterministically on the Hands-on Scala Programming build, when I try to use `-j 8` to build all 4 distributable copies of the book in parallel. Note sure if user error or Mill bug, but even if user error it would be nice to have a better error message

```
[#2] Exception in thread "main" java.lang.Exception: Duplicated item inserted into OrderedSet: mill.define.Task$Sequence@59405dfd
[#2]    at mill.api.AggWrapper$Agg$Mutable.append(AggWrapper.scala:56)
[#2]    at mill.api.AggWrapper$Agg$.$anonfun$from$1(AggWrapper.scala:43)
[#2]    at scala.collection.immutable.VectorStatics$.foreachRec(Vector.scala:1794)
[#2]    at scala.collection.immutable.Vector.foreach(Vector.scala:1800)
[#2]    at scala.collection.IterableOnceExtensionMethods$.foreach$extension(IterableOnce.scala:152)
[#2]    at mill.api.AggWrapper$Agg$.from(AggWrapper.scala:43)
[#2]    at mill.eval.Evaluator.parallelEvaluate(Evaluator.scala:238)
[#2]    at mill.eval.Evaluator.evaluate(Evaluator.scala:78)
[#2]    at mill.main.RunScript$.evaluate(RunScript.scala:202)
[#2]    at mill.main.RunScript$.$anonfun$evaluateTasks$1(RunScript.scala:188)
[#2]    at scala.util.Either.map(Either.scala:382)
[#2]    at mill.main.RunScript$.evaluateTasks(RunScript.scala:187)
[#2]    at mill.main.RunScript$.$anonfun$runScript$4(RunScript.scala:66)
[#2]    at ammonite.util.Res$Success.flatMap(Res.scala:62)
[#2]    at mill.main.RunScript$.runScript(RunScript.scala:65)
[#2]    at mill.main.MainRunner.$anonfun$runScript$1(MainRunner.scala:115)
[#2]    at mill.main.MainRunner.watchLoop2(MainRunner.scala:64)
[#2]    at mill.main.MainRunner.runScript(MainRunner.scala:90)
[#2]    at mill.MillMain$.main0(MillMain.scala:239)
[#2]    at mill.MillMain$.main(MillMain.scala:30)
[#2]    at mill.MillMain.main(MillMain.scala)
```
